primary/src/block_remover.rs

```diff
@@ -173,6 +188,9 @@ pub struct BlockRemover<PublicKey: VerifyingKey> {
     /// The persistent storage for payload markers from workers.
     payload_store: Store<(BatchDigest, WorkerId), PayloadToken>,
 
+    /// The Dag structure for managing the stored certificates
+    dag: Option<Arc<Dag<PublicKey>>>,
```

Comment:
I see that the approach we take with `Store` is to clone them every time we pass around to other structs. Here we are passing an `Arc` and I see the point (I guess we could go with the `clone` approach as well). What would be the best/recommended approach in general in this case? For example, would we want ideally to share the Stores via Arcs as well (or use copy for Dag)?